src/middlewared/middlewared/worker.py

```diff
@@ -131,6 +133,16 @@ def watch_parent():
     os._exit(1)
 
 
+def receive_environ():
+    def callback(*args, **kwargs):
+        environ_update(kwargs['fields'])
+
+    c = Client('ws+unix:///var/run/middlewared-internal.sock', py_exceptions=True)
+    c.subscribe('system.environ', callback)
+
+    environ_update(c.call('system.environ'))
```

Comment:
We dont have any problems of garbage collection getting rid of client as disconnecting do we? Since it should spawn a new thread waiting for responses...

Just making sure

Reply:
It should be a permanent connection that lives during worker entire lifetime